### PR TITLE
Feat/ci auto reqs

### PR DESCRIPTION
**Pull request by pgierz**
Automatically create the `requirements.txt` file, based upon imports in the code.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -62,8 +45,8 @@
     keywords="pyfesom2",
     name="pyfesom2",
     packages=["pyfesom2"],
-    #package_dir={"pyfesom2": "pyfesom2"},
-    #package_data={"": ["*.geojson"]},
+    # package_dir={"pyfesom2": "pyfesom2"},
```
Comment (pgierz) [inline setup.py:48]: Sorry about this one, my vim seems to automatically run black formatting whenever I save a Python file... not sure when/how I put that in there ;-)

**Comment by koldunovn** [comment]
So this will change the `requirements.txt` if something new is imported automatically? What about the versions of the packages that you put there now, will they be changed as well?

**Comment by pgierz** [comment]
Yup, as far as I understand it, this will pin the "newest" versions. 

Only thing I am not sure about is what to do we need a particular version of something, I'm reading through the documentation right now.

**Comment by pgierz** [comment]
Alright it just broke all of the checks. I may have accidentally messed up the conda installation, please don't merge yet.

Sorry...

File: setup.py (hunk 1 above)
Comment (koldunovn) [reply setup.py:48]: Happens to me all the time :)

File: requirements.txt [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-scipy==1.7.1
-seawater==3.3.4
-Shapely==1.7.1
-xarray==0.19.0
+cartopy
+cmocean
+dask
+fsspec
+joblib
+matplotlib
```
Comment (koldunovn) [inline requirements.txt:6]: @pgierz please note we have some requirements pinned to some versions in `requirements-py37.yml` :)

**Comment by pgierz** [comment]
Well, that was my thought of having everything getting detected automatically. Then you just import something in your new code, and it ends up automatically being in the requirements.txt. Next you can make a CI action that includes that list into whatever else might need it: `conda build`, for instance. In any case, I am nearly done with my testing. What we get when it's done:

* No more maintaining requirements files
* Make a tar.gz and a release on every tag push
* Automatic publishing to conda-forge on every tag push.

My only question so far is here:

https://github.com/conda-forge/pyfesom2-feedstock/blob/master/recipe/meta.yaml

https://github.com/conda-forge/pyfesom2-feedstock/blob/6b2bcfe04b484dba57573cbf0caffdcf9ec6160d/recipe/meta.yaml#L8-L10

There is a sha attached to source. Is that computed automatically, or did you need to supply it?

Also, there is a way to run conda build and conda upload directly from the CI. So I am not sure if we need to use this feedstock? But I am not sure.

**Comment by koldunovn** [comment]
To be honest, I don't remember how the SHA was created, but I suspect it's something that is done at the stage of PyPI package creation.

I believe without feedstock we will not be able to get to the `conda-forge` channel, and we don't want users to bother adding one more channel to their setup?

**Pull request closed without merging**